execution/execution.go

```diff
@@ -120,6 +108,29 @@ func newOperator(expr parser.Expr, storage *engstore.SelectorPool, opts *query.O
 		// before it can be non-nested. https://github.com/thanos-io/promql-engine/issues/39
 		for i := range e.Args {
 			switch t := e.Args[i].(type) {
+			case *parser.SubqueryExpr:
```

Comment:
Shouldn't this be a top-level `case` instead of having it here in the function args?